openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1117,6 +1118,53 @@ Cache-Control: no-store
 }
 ```
 
+# Callback Endpoint {#callback_endpoint}
+
+This endpoint is used to receive notification from the Wallet whether credential has been successfully received in the Wallet or not. Support for this endpoint is OPTIONAL.
+
+This endpoint can be used after the Credential Issuer has sent Credential Response or Batch Credential Response. It enables the Credential Issuer to take subsequent actions after issuance, depending on whether the credential has been accepted and successully stored by the Wallet, rejected by the Wallet, or errors and other unforeseen circumstances have occurred during the Wallet's processing.
+
+Communication with the Callback Endpoint MUST utilize TLS.
+
+## Callback from the Wallet
+
+The Wallet sends a callback to the Callback Endpoint by sending the following parameters in the entity-body of an HTTP POST request using the `application/json` media type.
+
+* `status`: REQUIRED. Status whether the credential issuance was successful or not. The value MUST be either `success` or `error`.
+* `error_description`: OPTIONAL. Human-readable ASCII [@!USASCII] text providing additional information, used to assist the Credential Issuer developer in understanding the error that occurred. Values for the `error_description`` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+Below is a non-normative example of a callback when credential issuance was successful:
```

Comment:
Interesting

There May be attacks that floods the vci server with fake errors using plain JSON http requests, to hijack in massive failures the issuances?

Reply:
Since It Is rest, is this endpoint protected?

Is there any value/element/property that allows a vci to filter out the anonymous and invalid, fake, requests?

Reply:
according to the latest commits this attack has been mitigated with the `callback_id`

Reply:
Apologies if this has already been discussed, but I think I would have expected this endpoint to require an access token as other credential issuer endpoints do - is there a reason not to use an access token in the case?

Reply:
Makes sense to apply the same protection (access token) as on the credential endpoint imho

Reply:
It makes sense @jogu @c2bo

Reply:
Agreed this needs to be protected using Access Token.

Reply:
+1 the wallet has an access token and should use it as with any other credential related endpoint (and also additional protection mechanisms, like DPoP)

Reply:
+1 agree with requiring an access token.

Reply:
+1 I honestly thought this was already included! good spot to make sure it's explicit

Reply:
thinking more about this.... scope and audience of AT is for the credential endpoint, not callback endpoint.... technically, AT used at the credential endpoint should not work at the callback endpoint..

Reply:
> thinking more about this.... scope and audience of AT is for the credential endpoint, not callback endpoint.... 

I don't think the current spec tightly defines what the AT is valid for.

I think it would make sense to say the access token includes this new endpoint as well. At least I can't see any reason not to.

Reply:
added
> The Wallet MUST present to the Callback Endpoint a valid Access Token issued at the Token Endpoint as defined in (#token_endpoint). Credential Issuer that requires this Callback MUST ensure Access Token issued by the Authorization Server is valid at the Callback Endpoint.